src/any_agent/tools/mcp/frameworks/agno.py

```diff
@@ -73,13 +75,47 @@ async def list_tools(self) -> list["AgnoMCPTools"]:
             headers=dict(self.mcp_tool.headers or {}),
         )
         sse_transport = await self._exit_stack.enter_async_context(client)
-        stdio, write = sse_transport
+        read_stream, write_stream = sse_transport
         kwargs = {}
         if self.mcp_tool.client_session_timeout_seconds:
             kwargs["read_timeout_seconds"] = timedelta(
                 seconds=self.mcp_tool.client_session_timeout_seconds
             )
-        client_session = ClientSession(stdio, write, **kwargs)  # type: ignore[arg-type]
+        client_session = ClientSession(read_stream, write_stream, **kwargs)  # type: ignore[arg-type]
+        session = await self._exit_stack.enter_async_context(client_session)
+        await session.initialize()
+        self._server = AgnoMCPTools(
+            session=session,
+            include_tools=list(self.mcp_tool.tools) if self.mcp_tool.tools else None,
+        )
+        return await super().list_tools()
+
+
+class AgnoMCPStreamableHttpConnection(AgnoMCPConnection):
+    mcp_tool: MCPStreamableHttp
+    _get_session_id_callback: Callable[[], str | None] | None = None
+
+    def get_session_id(self) -> str | None:
```

Comment:
The streamable http mcp implementation returns a function to retrieve the session id. I don't see that we'd be needing the session right now, but we may need to link e.g. a context to the specific mcp session this agent is using in this context.